[2025-07-25T11:13:32.611+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-07-25T11:13:32.637+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: elt_to_dwh.extract_task.crawl_news scheduled__2025-07-15T00:00:00+00:00 [queued]>
[2025-07-25T11:13:32.646+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: elt_to_dwh.extract_task.crawl_news scheduled__2025-07-15T00:00:00+00:00 [queued]>
[2025-07-25T11:13:32.647+0000] {taskinstance.py:2867} INFO - Starting attempt 1 of 1
[2025-07-25T11:13:32.664+0000] {taskinstance.py:2890} INFO - Executing <Task(PythonOperator): extract_task.crawl_news> on 2025-07-15 00:00:00+00:00
[2025-07-25T11:13:32.671+0000] {standard_task_runner.py:104} INFO - Running: ['airflow', 'tasks', 'run', 'elt_to_dwh', 'extract_task.crawl_news', 'scheduled__2025-07-15T00:00:00+00:00', '--job-id', '99', '--raw', '--subdir', 'DAGS_FOLDER/elt_to_dwh.py', '--cfg-path', '/tmp/tmprvgc6hfz']
[2025-07-25T11:13:32.674+0000] {standard_task_runner.py:105} INFO - Job 99: Subtask extract_task.crawl_news
[2025-07-25T11:13:32.675+0000] {logging_mixin.py:190} WARNING - /home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=3168) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-07-25T11:13:32.676+0000] {standard_task_runner.py:72} INFO - Started process 3178 to run task
[2025-07-25T11:13:32.716+0000] {task_command.py:467} INFO - Running <TaskInstance: elt_to_dwh.extract_task.crawl_news scheduled__2025-07-15T00:00:00+00:00 [running]> on host 74cfddf97016
[2025-07-25T11:13:32.791+0000] {taskinstance.py:3134} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='elt_to_dwh' AIRFLOW_CTX_TASK_ID='extract_task.crawl_news' AIRFLOW_CTX_EXECUTION_DATE='2025-07-15T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-07-15T00:00:00+00:00'
[2025-07-25T11:13:32.792+0000] {taskinstance.py:732} INFO - ::endgroup::
[2025-07-25T11:13:32.805+0000] {logging_mixin.py:190} INFO - Get news from 20250715T0000 to 20250715T2359
[2025-07-25T11:13:35.821+0000] {logging_mixin.py:190} INFO - Get news from 20250715T0000 to 20250715T1200
[2025-07-25T11:13:38.047+0000] {logging_mixin.py:190} INFO - Get news from 20250715T1201 to 20250715T2359
[2025-07-25T11:13:41.588+0000] {logging_mixin.py:190} INFO - The process of crawling 2356 news was successful
[2025-07-25T11:13:41.588+0000] {logging_mixin.py:190} INFO - Saving at /opt/airflow/data/raw/news/crawl_news-20250715.json
[2025-07-25T11:13:41.590+0000] {python.py:240} INFO - Done. Returned value was: None
[2025-07-25T11:13:41.595+0000] {taskinstance.py:341} INFO - ::group::Post task execution logs
[2025-07-25T11:13:41.595+0000] {taskinstance.py:353} INFO - Marking task as SUCCESS. dag_id=elt_to_dwh, task_id=extract_task.crawl_news, run_id=scheduled__2025-07-15T00:00:00+00:00, execution_date=20250715T000000, start_date=20250725T111332, end_date=20250725T111341
[2025-07-25T11:13:41.635+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 0
[2025-07-25T11:13:41.651+0000] {taskinstance.py:3901} INFO - 1 downstream tasks scheduled from follow-on schedule check
[2025-07-25T11:13:41.656+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
[2025-07-25T11:52:52.809+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-07-25T11:52:52.852+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: elt_to_dwh.extract_task.crawl_news scheduled__2025-07-15T00:00:00+00:00 [queued]>
[2025-07-25T11:52:52.873+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: elt_to_dwh.extract_task.crawl_news scheduled__2025-07-15T00:00:00+00:00 [queued]>
[2025-07-25T11:52:52.874+0000] {taskinstance.py:2867} INFO - Starting attempt 1 of 1
[2025-07-25T11:52:52.901+0000] {taskinstance.py:2890} INFO - Executing <Task(PythonOperator): extract_task.crawl_news> on 2025-07-15 00:00:00+00:00
[2025-07-25T11:52:52.913+0000] {standard_task_runner.py:104} INFO - Running: ['airflow', 'tasks', 'run', 'elt_to_dwh', 'extract_task.crawl_news', 'scheduled__2025-07-15T00:00:00+00:00', '--job-id', '166', '--raw', '--subdir', 'DAGS_FOLDER/elt_to_dwh.py', '--cfg-path', '/tmp/tmpiv3t9g0o']
[2025-07-25T11:52:52.917+0000] {standard_task_runner.py:105} INFO - Job 166: Subtask extract_task.crawl_news
[2025-07-25T11:52:52.921+0000] {logging_mixin.py:190} WARNING - /home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=4855) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-07-25T11:52:52.922+0000] {standard_task_runner.py:72} INFO - Started process 4865 to run task
[2025-07-25T11:52:52.964+0000] {task_command.py:467} INFO - Running <TaskInstance: elt_to_dwh.extract_task.crawl_news scheduled__2025-07-15T00:00:00+00:00 [running]> on host 74cfddf97016
[2025-07-25T11:52:53.076+0000] {taskinstance.py:3134} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='elt_to_dwh' AIRFLOW_CTX_TASK_ID='extract_task.crawl_news' AIRFLOW_CTX_EXECUTION_DATE='2025-07-15T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-07-15T00:00:00+00:00'
[2025-07-25T11:52:53.078+0000] {taskinstance.py:732} INFO - ::endgroup::
[2025-07-25T11:52:53.101+0000] {logging_mixin.py:190} INFO - Get news from 20250715T0000 to 20250715T2359
[2025-07-25T11:52:53.971+0000] {taskinstance.py:3313} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/scripts/elt_to_dwh/extract/crawl_news.py", line 40, in crawl_news
    raise KeyError(f"'feed' key not found in response. Full response:\n{json.dumps(response, indent=4)}")
KeyError: '\'feed\' key not found in response. Full response:\n{\n    "Information": "We have detected your API key as RSKHO6B8SELFWUYG and our standard API rate limit is 25 requests per day. Please subscribe to any of the premium plans at https://www.alphavantage.co/premium/ to instantly remove all daily rate limits."\n}'
[2025-07-25T11:52:53.982+0000] {taskinstance.py:1226} INFO - Marking task as FAILED. dag_id=elt_to_dwh, task_id=extract_task.crawl_news, run_id=scheduled__2025-07-15T00:00:00+00:00, execution_date=20250715T000000, start_date=20250725T115252, end_date=20250725T115253
[2025-07-25T11:52:54.011+0000] {taskinstance.py:341} INFO - ::group::Post task execution logs
[2025-07-25T11:52:54.011+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 166 for task extract_task.crawl_news ('\'feed\' key not found in response. Full response:\n{\n    "Information": "We have detected your API key as RSKHO6B8SELFWUYG and our standard API rate limit is 25 requests per day. Please subscribe to any of the premium plans at https://www.alphavantage.co/premium/ to instantly remove all daily rate limits."\n}'; 4865)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3006, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 274, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3161, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3185, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/scripts/elt_to_dwh/extract/crawl_news.py", line 40, in crawl_news
    raise KeyError(f"'feed' key not found in response. Full response:\n{json.dumps(response, indent=4)}")
KeyError: '\'feed\' key not found in response. Full response:\n{\n    "Information": "We have detected your API key as RSKHO6B8SELFWUYG and our standard API rate limit is 25 requests per day. Please subscribe to any of the premium plans at https://www.alphavantage.co/premium/ to instantly remove all daily rate limits."\n}'
[2025-07-25T11:52:54.061+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-07-25T11:52:54.083+0000] {taskinstance.py:3901} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-07-25T11:52:54.091+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
[2025-07-26T05:00:07.718+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-07-26T05:00:07.730+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: elt_to_dwh.extract_task.crawl_news scheduled__2025-07-15T00:00:00+00:00 [queued]>
[2025-07-26T05:00:07.739+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: elt_to_dwh.extract_task.crawl_news scheduled__2025-07-15T00:00:00+00:00 [queued]>
[2025-07-26T05:00:07.739+0000] {taskinstance.py:2867} INFO - Starting attempt 1 of 1
[2025-07-26T05:00:07.752+0000] {taskinstance.py:2890} INFO - Executing <Task(PythonOperator): extract_task.crawl_news> on 2025-07-15 00:00:00+00:00
[2025-07-26T05:00:07.758+0000] {standard_task_runner.py:104} INFO - Running: ['airflow', 'tasks', 'run', 'elt_to_dwh', 'extract_task.crawl_news', 'scheduled__2025-07-15T00:00:00+00:00', '--job-id', '51', '--raw', '--subdir', 'DAGS_FOLDER/elt_to_dwh.py', '--cfg-path', '/tmp/tmp95uxeil1']
[2025-07-26T05:00:07.760+0000] {standard_task_runner.py:105} INFO - Job 51: Subtask extract_task.crawl_news
[2025-07-26T05:00:07.761+0000] {logging_mixin.py:190} WARNING - /home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=1555) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-07-26T05:00:07.762+0000] {standard_task_runner.py:72} INFO - Started process 1565 to run task
[2025-07-26T05:00:07.791+0000] {task_command.py:467} INFO - Running <TaskInstance: elt_to_dwh.extract_task.crawl_news scheduled__2025-07-15T00:00:00+00:00 [running]> on host 3bc71adbf7e0
[2025-07-26T05:00:07.852+0000] {taskinstance.py:3134} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='elt_to_dwh' AIRFLOW_CTX_TASK_ID='extract_task.crawl_news' AIRFLOW_CTX_EXECUTION_DATE='2025-07-15T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-07-15T00:00:00+00:00'
[2025-07-26T05:00:07.853+0000] {taskinstance.py:732} INFO - ::endgroup::
[2025-07-26T05:00:07.865+0000] {logging_mixin.py:190} INFO - Get news from 20250715T0000 to 20250715T2359
[2025-07-26T05:00:10.600+0000] {logging_mixin.py:190} INFO - Get news from 20250715T0000 to 20250715T1200
[2025-07-26T05:00:12.783+0000] {logging_mixin.py:190} INFO - Get news from 20250715T1201 to 20250715T2359
[2025-07-26T05:00:16.019+0000] {logging_mixin.py:190} INFO - The process of crawling 2356 news was successful
[2025-07-26T05:00:16.019+0000] {logging_mixin.py:190} INFO - Saving at /opt/airflow/data/raw/news/crawl_news-20250715.json
[2025-07-26T05:00:16.021+0000] {python.py:240} INFO - Done. Returned value was: None
[2025-07-26T05:00:16.025+0000] {taskinstance.py:341} INFO - ::group::Post task execution logs
[2025-07-26T05:00:16.026+0000] {taskinstance.py:353} INFO - Marking task as SUCCESS. dag_id=elt_to_dwh, task_id=extract_task.crawl_news, run_id=scheduled__2025-07-15T00:00:00+00:00, execution_date=20250715T000000, start_date=20250726T050007, end_date=20250726T050016
[2025-07-26T05:00:16.084+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 0
[2025-07-26T05:00:16.097+0000] {taskinstance.py:3901} INFO - 1 downstream tasks scheduled from follow-on schedule check
[2025-07-26T05:00:16.103+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
